### ISSUE TITLE
[Bug]: TypeError: Object has been destroyed at main.js

### ISSUE THREAD

**Issue by iCaran** [FreeTubeApp/FreeTube#5553]
### Guidelines

- [X] I have encountered this bug in the [latest release of FreeTube](https://github.com/FreeTubeApp/FreeTube/releases).
- [X] I have encountered this bug in the [official downloads of FreeTube](https://github.com/FreeTubeApp/FreeTube#official-downloads).
- [X] I have searched the issue tracker for [open](https://github.com/FreeTubeApp/FreeTube/issues?q=is%3Aopen+is%3Aissue) and [closed](https://github.com/FreeTubeApp/FreeTube/issues?q=is%3Aissue+is%3Aclosed) issues that are similar to the bug report I want to file, without success.
- [X] I have searched the [documentation](https://docs.freetubeapp.io/) for information that matches the description of the bug I want to file, without success.
- [X] This issue contains only one bug.

### Describe the bug

My PC (Ubuntu 22.04.4 LTS) crashed, which is not unusual, its an old one with a very degraded battery, so it turns off when the power goes off. This time however I was using FreeTube when my pc shut down, and since I booted it back up, FreeTube won't start at all and I get this error:

![image](https://github.com/user-attachments/assets/056255c9-3454-46dd-b9ed-b23a07af50b6)

I have tried uninstalling and reinstalling FreeTube (v0.21.3) through Software Install, but the problem persists.

### Expected Behavior

It should work just fine like every other time my machine has crashed, this time however it broke FreeTube permanently somehow. Even Reinstalling doesn't seem to work

### Issue Labels

causes crash, usability issue

### FreeTube Version

v0.21.3

### Operating System Version

Ubuntu 22.04.4 LTS

### Installation Method

.deb

### Primary API used

Local API

### Nightly Build

- [ ] I have encountered this bug in the latest [nightly build](https://docs.freetubeapp.io/development/nightly-builds).

**Comment by absidue** [comment]
During the crash FreeTube didn't have time to properly shutdown so it thinks it is still running. Close FreeTube, go to `~/.config/FreeTube` and delete the file called `lockfile`, that should fix your problem.

**Comment by iCaran** [comment]
Do you mean the `SingletonLock` file?

![image](https://github.com/user-attachments/assets/ef0fbc40-fdd3-4ed1-a70a-ee6bb4014d4c)

**Comment by absidue** [comment]
Looks like it has a different name on Linux but yes try deleting all three singleton files (on Windows it's just one).

**Comment by iCaran** [comment]
I deleted it and restared Freetube, same error, the files are labled as `Link (broken)` again

**Comment by TheodorSmall** [comment]
I found out the following:
By default on Linux, FreeTube seems to create these three Singleton* files:
- SingletonCookie and SingletonLock, which are symbolic links, that seem to use the filename they're referencing as a storage (therefore no such file exists and the type is displayed as "broken" link, this is intentional)
- SingletonLock stores the process ID (PID) and the user running the app; the former is temporary and is renewed when FreeTube restarts, the latter will be constant (because the storage path depends on the user)
  However, FreeTube should be able to recognize an invalid PID and ignore it
- SingletonCookie stores a decimal number, however I can't tell what exact information is stored here
- SingletonSocket is a valid symlink to a file in a *temporary* directory (/tmp/<random_string>/SingletonSocket), which is a socket file; it is deleted automatically during the boot process
  This will also be broken on a crash and/or reboot, but that shouldn't be an issue either

The fact these files persist after a crash doesn't even seem to be the issue, since their deletion has no effect. Something else seems to be broken, which causes FreeTube to crash every time, regenerating the Singleton* files.

Please also note that any reinstallation process won't delete any configuration files or user data, which in this case are located in .config/FreeTube. Assuming the issue lies in this directory and some invalid file causes the crash, there are generally two possible solutions, while in both cases you should back up the complete ~/.config/FreeTube folder, because that's where all your personal data and your settings are stored. To copy them, you can use the following command or use the GUI if you prefer:
`cp -r ~/.config/FreeTube `

- Solution 1: Find the corrupted file(s), repair or delete them (depending on their relevance)
- Solution 2: Extract your personal data (settings, history, profiles/subscriptions, playlists), then delete .config/FreeTube completely (only after a backup!), or move/rename it

**Comment by TheodorSmall** [comment]
While FreeTube is not running, there would be no reason for a _history.db~_ file to exist, maybe try renaming that (e.g. to _history2.db~_), in case that solves the issue, and no relevant data is lost, you can delete it.

I suppose the relevant files you might want to keep for solution 2 are the following:
- history.db
- playlists.db
- profiles.db
- settings,db

However, don't rely on that and better keep a complete backup of .config/FreeTube for some time.

**Comment by absidue** [comment]
None of your user data is corrupted.
1. Copy the 4 files mentioned here to a different folder: https://docs.freetubeapp.io/usage/data-location/
2. Delete everything else in that folder
3. Copy those 4 files back
4. Start FreeTube it should work fine again

On Windows the "is already running" stuff happens with just one file called `lockfile`, however by the sounds of it in typical Linux fashion, it's a lot more complicated there, so instead of trying to figure out how all of the "is already running" stuff works on Linux, it's better to just delete everything that isn't your user data and start again from there.

**Comment by absidue** [comment]
I have to say I'm surprised that you never had any problems every other time that your computer crashed with FreeTube open, you definitely just got lucky the other times, because when the app doesn't shutdown properly then you should expect stuff to be broken.

**Comment by iCaran** [comment]
I did exactly this, i deleted the Freetube folder from .config entirely, restarted freetube, it worked, but was empty and new, so i copied back just the 4 files into the newly created Freetube directory in .config again, and guess what... Same Crash, Same Error Message AGAIN, 

just the 4 db files caused this

**Comment by absidue** [comment]
In that case I'll close this as a duplicate of the existing open issue, as the cause remains a mystery because looking at the code there is no connection between the code that produces the error and the databases.

duplicate of #4764

**Comment by iCaran** [comment]
Well any suggestions as to what I could do or try tho? Do i really have to save goodbye to all my saved playlists, subscriptions, and history?

**Comment by TheodorSmall** [comment]
You can try to only copy one of the 4 files back at a time to identify which is causing the crash. To be sure, you can remove that corrupted file to check the remaining ones as well. From there, you can continue to use the valid files, and we need to figure out how to save your user data from the corrupted ones.
Be sure to do backups regularly from now on (see the settings -> data, still the settings.db needs to be copied manually).

**Comment by iCaran** [comment]
Sorry for the delayed reply, but I think I have identified the cause. It is profiles.db, i think in an improper shutdown, it gets corrupted.

![Screenshot from 2024-08-14 01-05-12](https://github.com/user-attachments/assets/0053f5bc-5c07-45d0-9293-45fc86aad6ac)
![Screenshot from 2024-08-14 01-05-22](https://github.com/user-attachments/assets/37b15bf8-eeb0-4207-b29a-0e31d5d56b7e)

See the weird and very long line of `\00\` at the end? I deleted that line, had imported the rest 3 db files as is prior to profiles.db, so i knew the final profiles.db file was the culprit.

Once i removed that weird corruption at the end, and saved it, Freetube works normally again!

**Comment by iCaran** [comment]
I suggest a feature implementation which automatically checks for corruption in profiles.db when such a crash happens, and automatically clean the file.

**Comment by wcarss** [comment]
just had [a similar issue resolved](https://github.com/FreeTubeApp/FreeTube/issues/4762) by a malformed json object in subscription-cache.db -- this seems like it may be possible in any .db file and should likely be checked for / more gracefully handled in any of them.

It was additionally causing [our subscriptions not to load](https://github.com/FreeTubeApp/FreeTube/issues/6259) and just spin forever, which is also now resolved by removing that file.
